### PR TITLE
CNI: add host-side interface info to cni.Result

### LINKED ISSUE #24899: cilium CNI cannot be used in combination with `bandwidth` CNI
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

In my testing environment, I tested the scenario of using `cilium` and `bandwidth` CNI together(we cannot use cilium `bandwidthManager` because our os kernel version is too low to meet the requirements.). Then we found that an error occured when `bandwith` CNI was called: `no veth peer of container interface found in host ns`.

After reivewing the `bandwith` CNI source code, the reason for this error is that the result returned by cilium CNI does not meet the following containtion.
https://github.com/containernetworking/plugins/blob/4a6147a1552064af80b4f7567b30c5174153c62a/plugins/meta/bandwidth/main.go#L153-L160
```go
for _, iface := range interfaces {
    if iface.Sandbox == "" && iface.Name == link.Attrs().Name {
	return iface, nil
    }
}
```

The official documentation of CNI requires that the `interfaces` field should include information about both container-side interface and host-side interface. However, currently cilium only provides the container-side interface information.

https://www.cni.dev/docs/spec/#section-5-result-types
- `interfaces`: An array of all interfaces created by the attachment, including any host-level interfaces:
    - `name`: The name of the interface.
    - `mac`: The hardware address of the interface (if applicable).
    - `sandbox`: The isolation domain reference (e.g. path to network namespace) for the interface, or empty if on the host. For interfaces created inside the container, this should be the value passed via `CNI_NETNS`.

some test code:
```diff
+       log.Println("interfaces length:", len(interfaces))
        for _, iface := range interfaces {
+               log.Println("iface name:", iface.Name)
+               log.Println("sandbox: ", iface.Sandbox)
+               log.Println("link.Attrs().Name: ", link.Attrs().Name)
                if iface.Sandbox == "" && iface.Name == link.Attrs().Name {
                        return iface, nil
                }
        }
```

output:
```
2023/04/13 07:20:51 interfaces length: 1
2023/04/13 07:20:51 iface name: eth0
2023/04/13 07:20:51 sandbox:  /var/run/netns/cni-882cb34f-9dd9-6014-556b-724e5d2e383d
2023/04/13 07:20:51 link.Attrs().Name:  lxc1986c9428372
```

### Cilium Version

1.13.0

### Kernel Version

5.15.0-69-generic

### Kubernetes Version

Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.3", GitCommit:"9e644106593f3f4aa98f8a84b23db5fa378900bd", GitTreeState:"clean", BuildDate:"2023-03-15T13:40:17Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.3", GitCommit:"9e644106593f3f4aa98f8a84b23db5fa378900bd", GitTreeState:"clean", BuildDate:"2023-03-15T13:33:12Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nayihz**
Fixes: https://github.com/cilium/cilium/issues/24899

```release-note
Add host-side interface info to cni.Result, which allows bandwidth CNI to work with Cilium
```

### CODE REVIEW

File: plugins/cilium-cni/main.go [hunk 1]
```diff
@@ -534,6 +539,8 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 		if err != nil {
 			return fmt.Errorf("unable to prepare IP addressing for '%s': %s", ep.Addressing.IPV6, err)
 		}
+		// Add to the result the Interface as index of Interfaces
+		ipConfig.Interface = cniTypesV1.Int(len(res.Interfaces))
```
Comment (squeed) [inline plugins/cilium-cni/main.go:543]: nit: this comment isn't particularly correct. Instead, can you put something more descriptive?

e.g. "set the addresses interface index to that of the container-side veth"?

**Review by squeed — APPROVED** [review]
One minor request, then looks good!

**Comment by borkmann** [comment]
> ```
> Add host-side interface info to cni.Result, which allows bandwidth CNI to work with Cilium
> ```

Off topic and unrelated to the fix, have you considered using Cilium's bandwidth manager? https://docs.cilium.io/en/stable/network/kubernetes/bandwidth-manager/

**Review by tommyp1ckles — APPROVED** [review]

**Comment by tommyp1ckles** [comment]
mirroring +1 from last pr which was closed: https://github.com/cilium/cilium/pull/24901

**Comment by gandro** [comment]
Marking for backport since this is a bug fix.

**Pull request merged**